### PR TITLE
FBUS Support

### PR DESCRIPTION
**Pull request by robthomson**
Support for FBUS Protocol.

### CODE REVIEW

**Review by rotorflight — CHANGES_REQUESTED** [review]
I think we should call it fbus.c on the top level.
Likewise, unless things

File: make/source.mk [hunk 1]
```diff
@@ -126,6 +126,7 @@ COMMON_SRC = \
             rx/sumh.c \
             rx/xbus.c \
             rx/fport.c \
+            rx/fport2.c \
```
Comment (rotorflight) [inline make/source.mk:129]: rx/fbus.c

File: src/main/build/debug.c [hunk 2]
```diff
@@ -60,6 +60,7 @@ const char * const debugModeNames[DEBUG_COUNT] = {
     DEBUG_NAME(MAX7456_SPICLOCK),
     DEBUG_NAME(SBUS),
     DEBUG_NAME(FPORT),
+    DEBUG_NAME(FPORT2),
```
Comment (rotorflight) [inline src/main/build/debug.c:63]: New debug items _MUST_ go to the end of the list

File: src/main/build/debug.h [hunk 3]
```diff
@@ -71,6 +71,7 @@ typedef enum {
     DEBUG_MAX7456_SPICLOCK,
     DEBUG_SBUS,
     DEBUG_FPORT,
+    DEBUG_FPORT2,
```
Comment (rotorflight) [inline src/main/build/debug.h:74]: New debug items _MUST_ go to the end of the list

**Review by rotorflight — CHANGES_REQUESTED** [review]
Let's call it fbus, as that's what FrSky is calling it

**Comment by robthomson** [comment]
> Let's call it fbus, as that's what FrSky is calling it

So this is an issue we discussed this end quite a bit.

I can rename it.. But we also have fport 2 as part of the same tree.

And to add to the mix frsky mark their fbus receivers with labels saying fport!

It's  a mess on their naming.


I can for sure change this to be fbus.

I wonder maybe if we do a full split and do fport 2 in its own source files?

File: src/main/build/debug.h (hunk 3 above)
Comment (robthomson) [reply src/main/build/debug.h]: Doe

File: src/main/build/debug.c (hunk 2 above)
Comment (robthomson) [reply src/main/build/debug.c]: Done

File: src/main/cli/settings.c [hunk 4]
```diff
@@ -229,10 +229,12 @@ static const char * const lookupTableSerialRX[] = {
     "CRSF",
     "SRXL",
     "CUSTOM",
-    "FPORT",
+    "FPORT",
```
[the remark this answers is not in the document]
Comment (robthomson) [reply src/main/cli/settings.c]: Fixed

File: make/source.mk (hunk 1 above)
Comment (robthomson) [reply make/source.mk]: All done

**Comment by rotorflight** [comment]
Integrated manually

**Pull request closed without merging**